scraper.py

```diff
@@ -1,156 +1,149 @@
-import csv
 import datetime
 import os
 import random
 import requests
 import time
 from dateutil.relativedelta import relativedelta
 from polling_bot.brain import SlackClient
-from sopn_publish_date import StatementPublishDate, Country
-from sopn_publish_date.election_ids import (
+from uk_election_timetables.calendars import Country
+from uk_election_timetables.election_ids import from_election_id
+from uk_election_timetables.election_ids import (
     InvalidElectionIdError,
     NoSuchElectionTypeError,
 )
 
 # hack to override sqlite database filename
 # see: https://help.morph.io/t/using-python-3-with-morph-scraperwiki-fork/148
-os.environ['SCRAPERWIKI_DATABASE_NAME'] = 'sqlite:///data.sqlite'
+os.environ["SCRAPERWIKI_DATABASE_NAME"] = "sqlite:///data.sqlite"
 import scraperwiki
 
 
 UPDATE_FREQUENCY = relativedelta(weeks=1)
 ELECTIONS_IN_SCOPE = relativedelta(months=1)
 MAX_OUTPUT_LINES = 30
 try:
-    SLACK_WEBHOOK_URL = os.environ['MORPH_ORINOCO_SLACK_WEBHOOK_URL']
+    SLACK_WEBHOOK_URL = os.environ["MORPH_ORINOCO_SLACK_WEBHOOK_URL"]
 except KeyError:
     SLACK_WEBHOOK_URL = None
 NOW = datetime.datetime.now()
-SOPN_PUBLISH_DATE = StatementPublishDate()
+YNR_BASE = "https://candidates.democracyclub.org.uk"
 
 
 def init():
-    scraperwiki.sql.execute("""
+    table_info = scraperwiki.sql.execute("PRAGMA table_info(data);")
+    if table_info.get("data"):
+        if "post_id" in [row[1] for row in table_info["data"]]:
+            scraperwiki.sql.execute("DROP TABLE data;")
```

Comment:
This is quite a crude "migration".

The only thing we really use the DB for is working out when we last sent a notification so we can run every day but only post once a week. The actual difference in state between runs is basically irrelevant. As such, the only impact of deleting all the data is we'll "reset the counter" and send slack notifications next time we deploy even if we last posted to slack yesterday.